src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -882,9 +886,6 @@ def match(self, segments, parse_context):
                 if self.exclude.match(segments, parse_context=ctx):
                     return MatchResult.from_unmatched(segments)
 
-        if not elem:  # pragma: no cover
-            raise ValueError(f"Null Element returned! _elements: {self._elements!r}")
-
```

Comment:
Definitely want to remove this one? I'm usually OK leaving in `no cover`s for exceptions.

Reply:
I've removed it because the value it refers to should never be `None` - see the type hints I've added to `_get_elem` above. It calls `Dialect.ref()` which will raise an exception rather than return `None`.